plugins/funder.c

```diff
@@ -586,6 +586,138 @@ static void json_channel_open_failed(struct command *cmd,
+	json_add_amount_sat_only(stream, "reserve_tank",
+				 policy->reserve_tank);
+	json_add_num(stream, "fuzz_percent", policy->fuzz_factor);
+	json_add_num(stream, "fund_probability", policy->fund_probability);
+}
+
+static struct command_result *
+param_funder_opt(struct command *cmd, const char *name,
+		 const char *buffer, const jsmntok_t *tok,
+		 enum funder_opt **opt)
+{
+	char *opt_str, *err;
+
+	*opt = tal(cmd, enum funder_opt);
+	opt_str = tal_strndup(cmd, buffer + tok->start,
+			      tok->end - tok->start);
+
+	err = funding_option(opt_str, *opt);
+	if (err)
+		return command_fail_badparam(cmd, name, buffer, tok, err);
+
+	return NULL;
+}
+
+
+static struct command_result *
+param_policy_mod(struct command *cmd, const char *name,
+		 const char *buffer, const jsmntok_t *tok,
+		 u64 **mod)
+{
+	struct amount_sat sats;
+	char *arg_str, *err;
+
+	*mod = tal(cmd, u64);
+	arg_str = tal_strndup(cmd, buffer + tok->start,
+			      tok->end - tok->start);
+
+	err = u64_option(arg_str, *mod);
+	if (err) {
+		if (!parse_amount_sat(&sats, arg_str, strlen(arg_str)))
+			return command_fail_badparam(cmd, name,
+						     buffer, tok, err);
+
+		**mod = sats.satoshis; /* Raw: convert to u64 */
+	}
+
+	return NULL;
+}
+
+static struct command_result *
+json_funderupdate(struct command *cmd,
+		  const char *buf,
+		  const jsmntok_t *params)
+{
+	struct json_stream *res;
+	struct amount_sat *min_their_funding, *max_their_funding,
+			  *per_channel_min, *per_channel_max,
+			  *reserve_tank;
+	u32 *fuzz_factor, *fund_probability;
+	u64 *mod;
+	enum funder_opt *opt;
+	const char *err;
+
+	if (!param(cmd, buf, params,
+		   p_opt("policy", param_funder_opt, &opt),
+		   p_opt("policy_mod", param_policy_mod, &mod),
+		   p_opt("min_their_funding", param_sat, &min_their_funding),
+		   p_opt("max_their_funding", param_sat, &max_their_funding),
+		   p_opt("per_channel_min", param_sat, &per_channel_min),
+		   p_opt("per_channel_max", param_sat, &per_channel_max),
+		   p_opt("reserve_tank", param_sat, &reserve_tank),
+		   p_opt("fuzz_percent", param_number, &fuzz_factor),
+		   p_opt("fund_probability", param_number, &fund_probability),
+		   NULL))
+		return command_param_failed();
+
+	if (opt)
+		current_policy.opt = *opt;
+	if (mod)
+		current_policy.mod = *mod;
+	if (min_their_funding)
+		current_policy.min_their_funding = *min_their_funding;
+	if (max_their_funding)
+		current_policy.max_their_funding = *max_their_funding;
+	if (per_channel_min)
+		current_policy.per_channel_min = *per_channel_min;
+	if (per_channel_max)
+		current_policy.per_channel_max = *per_channel_max;
+	if (reserve_tank)
+		current_policy.reserve_tank = *reserve_tank;
+	if (fuzz_factor)
+		current_policy.fuzz_factor = *fuzz_factor;
+	if (fund_probability)
+		current_policy.fund_probability = *fund_probability;
+
+	err = funder_check_policy(&current_policy);
+	if (err)
+		return command_done_err(cmd, JSONRPC2_INVALID_PARAMS,
+					err, NULL);
```

Comment:
I think you need to modify a copy, then assign it back to current_policy at the end. 

Otherwise, setting a bad value causes an error, but sticks around...